### ISSUE TITLE
I get an error when posting

### ISSUE THREAD

**Issue by opdrift** [Saturate/mock-api-middleware#1]
[MockApi] Serving POST request: /api/subscription/createsingle 1 B
Error: Can't set headers after they are sent.
    at ServerResponse.OutgoingMessage.setHeader (_http_outgoing.js:335:11)
    at ServerResponse.res.setHeader (D:\Projects\Website\node_modules\gulp-connect\node_modules\connect\lib\patch.js:134:22)
    at D:\Projects\Website\node_modules\serve-index\index.js:104:11
    at call (D:\Projects\Website\node_modules\connect\index.js:239:7)
    at next (D:\Projects\Website\node_modules\connect\index.js:183:5)
    at MockApiMiddleware.handle (D:\Projects\Website\node_modules\mock-api-middleware\index.js:75:4)
    at call (D:\Projects\Website\node_modules\connect\index.js:239:7)
    at next (D:\Projects\Website\node_modules\connect\index.js:183:5)
    at next (D:\Projects\Website\node_modules\connect\index.js:161:14)
    at next (D:\Projects\Website\node_modules\connect\index.js:161:14)

**Comment by moji29nabe** [comment]
Im also facing the same issue.
Using mock-api-middleware 0.0.3 and axios 0.15.3.

**Comment by Saturate** [comment]
I had forgotten this issue, I am sorry.

Will take a look at it soon :-)

And also add some UNIT Tests to make sure everything works correctly.

**Comment by Saturate** [comment]
Can't seem to recreate it, can you give me some code that creates this error?

**Comment by wxt2005** [comment]
@Saturate You can reproduce it by enabling browsersync's directory option.

```js
  browserSync.init({
    //...
    server: {
        // ...
        directory: true,
        // ...
    }
  });
```

I found this problem relates to server-index middleware, it treats the mock request as a querying for listing files.

**Comment by Saturate** [comment]
Thanks @wxt2005, I'll take a look at it when I have the time. If you feel adventurous I would happily accept PR's

**Comment by horizon0514** [comment]
Because you try to call next() function after ` res.end` or `res.send`
Delete the `next()`, it should works

**Comment by Saturate** [comment]
Should be fixed in #3 with `0.1.4`
